pkg/service/controller/servingcert/secret_updating_controller.go

```diff
@@ -234,7 +236,13 @@ func (sc *ServiceServingCertUpdateController) syncSecret(key string) error {
 	dnsName := secret.Annotations[ServiceNameAnnotation] + "." + secret.Namespace + ".svc"
 	fqDNSName := dnsName + "." + sc.dnsSuffix
 	certificateLifetime := 365 * 2 // 2 years
-	servingCert, err := sc.ca.MakeServerCert(sets.NewString(dnsName, fqDNSName), certificateLifetime)
+	servingCert, err := sc.ca.MakeServerCert(
+		sets.NewString(dnsName, fqDNSName),
+		certificateLifetime,
+		extensions.ServiceServerCertificateExtension(&kapi.Service{
+			ObjectMeta: kapi.ObjectMeta{UID: types.UID(secret.Annotations[ServiceUIDAnnotation])},
```

Comment:
> @deads2k can you say why you didn't re-look up the service and verify the both way here? Is it happening earlier?

Previously, the name was signed for, but the UID was not signed for.  The worst someone could do is make a set of non-sensical metadata that made the controller do nothing. Now that you're signing this information, you need a stronger guarantee.